src/common/permission_utils.py

```
@@ -192,9 +193,47 @@ def get_plugin_dir_mode() -> int:
+    Used when normal removal fails due to root-owned files (e.g., __pycache__
+    directories created by the root ledmatrix service). Requires a sudoers rule
+    allowing the current user to run 'rm -rf' on the specific path.
+
+    Args:
+        path: Directory path to remove
+
+    Returns:
+        True if removal succeeded, False otherwise
+    """
+    try:
+        result = subprocess.run(
```

Ruff (0.15.0)

[error] `subprocess` call: check for execution of untrusted input

(S603)